File: TopInterview150/Hashmap/ValidAnagram/solution.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map<char, int> counts;
        for (char c: s)
        {
            if (counts.find(c) != counts.end())
            {
                counts[c] += 1;
            }
            else counts[c] = 1;
        }
        for (char c: t)
        {
            if (counts.find(c) == counts.end())
            {
                return false;
            }
            counts[c]--;
        }
        for (const auto& count : counts)
        {
            if (count.second != 0)
                return false;
        }
        return true;
    }
};

File: TopInterview150/Hashmap/ValidAnagram/solution.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        count = {}
        for char in s:
            count[char] = count.setdefault(char, 0) + 1
        for char in t:
            count[char] = count.setdefault(char, 0) - 1
        for value in count.values():
            if value != 0:
                return False
        return True